### PR TITLE
RCF-32, 37, 131

### PR DESCRIPTION
**Pull request by Piyush7034**
Added offline login and onboarded and non-onboarded users dashboard page

### CODE REVIEW

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/config/SessionManager.java [hunk 1]
```diff
@@ -57,6 +57,7 @@ public void saveAuthToken(@NonNull String token) throws Exception {
         editor.putBoolean(IS_DEFAULT, roles.contains("Default"));
         editor.putBoolean(IS_OFFICER, roles.contains("REGISTRATION_OFFICER"));
         editor.apply();
+        return roles;
```
Comment (ase-101) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/config/SessionManager.java:60]: As we store complete role details in sharedPreferences, is it required to return the list of roles from here?

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @ColumnInfo(name = "pwd")
+    private String pwd;
+
+    @ColumnInfo(name = "pwd_expiry_dtimes")
+    private String pwdExpiryDtimes;
+
+    @ColumnInfo(name = "status_code")
+    private String statusCode;
+
+    @ColumnInfo(name = "lang_code")
```
Comment (ase-101) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java:33]: langCode column is not required, lets remove it.

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    @NonNull
+    @PrimaryKey
+    @ColumnInfo(name = "usr_id")
+    private String usrId;
+
+    @ColumnInfo(name = "pwd")
+    private String pwd;
+
+    @ColumnInfo(name = "pwd_expiry_dtimes")
+    private String pwdExpiryDtimes;
```
Comment (ase-101) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java:28]: We do not expire passwords in registration clients, this column is not required

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserDetail.java [hunk 4]
```diff
@@ -31,6 +29,9 @@ public class UserDetail implements Serializable {
 	@ColumnInfo(name = "reg_id")
```
Comment (ase-101) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserDetail.java:29]: Also lets remove reg_id from user_detail table as its no more used.

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java [hunk 5]
```diff
@@ -61,7 +73,65 @@ public boolean isActiveUser(String userId) {
         return userDetailDao.getUserDetail(userId) != null;
     }
 
+public UserDetail getUserDetailByUserId(String userId){
+
+        return userDetailDao.getUserDetail(userId);
+    }
```
Comment (ase-101) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java:79]: provide atleast one new line between every methods

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/LoginService.java [hunk 6]
```diff
@@ -54,7 +74,8 @@ public void saveAuthToken(String authResponse) throws Exception {
         byte[] decodedBytes = CryptoUtil.base64decoder.decode(cryptoResponseDto.getValue());
         try {
             JSONObject jsonObject = new JSONObject(new String(decodedBytes));
-            this.sessionManager.saveAuthToken(jsonObject.getString("token"));
+            List<String> roles=this.sessionManager.saveAuthToken(jsonObject.getString("token"));
+            return roles;
```
Comment (ase-101) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/LoginService.java:78]: when the roles are already part of shreadPreferences why do we need the list of roles to be returned.

File: android/packetmanager/src/main/java/io/mosip/registration/packetmanager/util/HMACUtils2.java [hunk 7]
```diff
@@ -19,10 +26,90 @@ public static byte[] generateHash(byte[] bytes) throws NoSuchAlgorithmException
...
+        return Base64.getDecoder().decode(data);
+    }
+
+    /*
+     * No object initialization.
+     */
+    private HMACUtils2() {
+    }
+
+    private static String encode(String password, byte[] salt) {
```
Comment (ase-101) [inline android/packetmanager/src/main/java/io/mosip/registration/packetmanager/util/HMACUtils2.java:84]: Where are we using this method?

File: assets/images/myself.jpg
[the remark this answers is not in the document]
Comment (Piyush7034) [reply assets/images/myself.jpg:1]: Deleted test files and unused files

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java (hunk 2 above)
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java]: Removed langCode column from UserPassword table

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java (hunk 3 above)
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserPassword.java]: removed password expiry field from UserPassword

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserDetail.java [hunk 8]
```diff
@@ -31,6 +29,9 @@ public class UserDetail implements Serializable {
...
 	private String regId;
 
+	@ColumnInfo(name = "salt")
+	private String salt;
```
[the remark this answers is not in the document]
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserDetail.java]: Removed salt from user_detail and stored in user_pwd

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserDetail.java (hunk 4 above)
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/entity/UserDetail.java]: Removed reg_id from user_detail table

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java (hunk 5 above)
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java:77]: Code formatting changed where required.

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/config/SessionManager.java (hunk 1 above)
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/config/SessionManager.java:60]: To be used as it is for now

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/LoginService.java (hunk 6 above)
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/LoginService.java:78]: To be used as it is for now

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/config/SessionManager.java (hunk 1 above)
Comment (ase-101) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/config/SessionManager.java:60]: This work will be taken in the next sprint.

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/LoginService.java (hunk 6 above)
Comment (ase-101) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/LoginService.java:78]: This work should be taken in the next sprint.

**Pull request merged**